core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -410,36 +409,29 @@ public VersionNumber getRequiredVersion() {
         }
     }
 
-    private static final List<DetachedPlugin> DETACHED_LIST = Collections.unmodifiableList(Arrays.asList(
-            new DetachedPlugin("maven-plugin", "1.296", "1.296"),
-            new DetachedPlugin("subversion", "1.310", "1.0"),
-            new DetachedPlugin("cvs", "1.340", "0.1"),
-            new DetachedPlugin("ant", "1.430.*", "1.0"),
-            new DetachedPlugin("javadoc", "1.430.*", "1.0"),
-            new DetachedPlugin("external-monitor-job", "1.467.*", "1.0"),
-            new DetachedPlugin("ldap", "1.467.*", "1.0"),
-            new DetachedPlugin("pam-auth", "1.467.*", "1.0"),
-            new DetachedPlugin("mailer", "1.493.*", "1.2"),
-            new DetachedPlugin("matrix-auth", "1.535.*", "1.0.2"),
-            new DetachedPlugin("windows-slaves", "1.547.*", "1.0"),
-            new DetachedPlugin("antisamy-markup-formatter", "1.553.*", "1.0"),
-            new DetachedPlugin("matrix-project", "1.561.*", "1.0"),
-            new DetachedPlugin("junit", "1.577.*", "1.0"),
-            new DetachedPlugin("bouncycastle-api", "2.16.*", "2.16.0"),
-            new DetachedPlugin("command-launcher", "2.86.*", "1.0")
-    ));
+    /** Record of which plugins which removed from core and when. */
+    private static final List<DetachedPlugin> DETACHED_LIST;
 
     /** Implicit dependencies that are known to be unnecessary and which must be cut out to prevent a dependency cycle among bundled plugins. */
-    private static final Set<String> BREAK_CYCLES = new HashSet<String>(Arrays.asList(
-            "script-security/matrix-auth",
-            "script-security/windows-slaves",
-            "script-security/antisamy-markup-formatter",
-            "script-security/matrix-project",
-            "script-security/bouncycastle-api",
-            "script-security/command-launcher",
-            "credentials/matrix-auth",
-            "credentials/windows-slaves"
-    ));
+    private static final Set<String> BREAK_CYCLES;
+
+    static {
+        List<DetachedPlugin> detached = new ArrayList<>();
+        try (InputStream is = ClassicPluginStrategy.class.getResourceAsStream("/jenkins/split-plugins.txt")) {
```

Comment:
It would be great to handle `null` resources, just in case the resource is missing due to whatever reason (repackaged WAR, classloader issues, etc.). Currently the code will fail with NPE

Reply:
That is fine with me. If someone deletes resources from `jenkins-core.jar`, they will get an error.